sklearn/linear_model/_glm/glm.py

```diff
@@ -20,13 +23,519 @@
+            if self.verbose:
+                print(f"Newton iter={self.iteration}")
+            # 1. Update hessian and gradient
+            self.update_gradient_hessian(X, y, sample_weight)
+
+            # TODO:
+            # if iteration == 1:
+            # We might stop early, e.g. we already are close to the optimum,
+            # usually detected by zero gradients at this stage.
+
+            # 2. Inner solver
+            #    Calculate Newton step/direction
+            #    This usually sets self.coef_newton.
+            self.inner_solve()
+
+            # 3. Backtracking line search
+            #    This usually sets self.coef_old, self.coef, self.loss_value_old
+            #    self.loss_value, self.gradient_old, self.gradient,
+            #    self.raw_prediction.
+            self.line_search(X, y, sample_weight)
+
+            # 4. Check convergence
+            #    Sets self.converged.
+            self.check_convergence(
+                X=X,
+                y=y,
+                sample_weight=sample_weight,
+            )
+
+            # 5. Next iteration
+            self.iteration += 1
+
+        if not self.converged:
+            warnings.warn(
+                "Newton solver did not converge after"
+                f" {self.iteration - 1} iterations.",
+                ConvergenceWarning,
+            )
+
+        self.iteration -= 1
+        self.finalize(X, y, sample_weight)
+        return self.coef
+
+
+class CholeskyNewtonSolver(NewtonSolver):
+    """Cholesky based Newton solver.
+
+    Inner solver for finding the Newton step H w_newton = -g uses Cholesky based linear
+    solver.
+    """
+
+    def setup(self, X, y, sample_weight):
+        super().setup(X=X, y=y, sample_weight=sample_weight)
+
+        n_dof = X.shape[1]
+        if self.linear_loss.fit_intercept:
+            n_dof += 1
+        self.gradient = np.empty_like(self.coef)
+        self.hessian = np.empty_like(self.coef, shape=(n_dof, n_dof))
+
+    def update_gradient_hessian(self, X, y, sample_weight):
+        self.linear_loss.gradient_hessian(
+            coef=self.coef,
+            X=X,
+            y=y,
+            sample_weight=sample_weight,
+            l2_reg_strength=self.l2_reg_strength,
+            n_threads=self.n_threads,
+            gradient_out=self.gradient,
+            hessian_out=self.hessian,
+            raw_prediction=self.raw_prediction,  # this was updated in line_search
+        )
+
+    def inner_solve(self):
+        try:
+            self.coef_newton = scipy.linalg.solve(
+                self.hessian, -self.gradient, check_finite=False, assume_a="sym"
+            )
+        except np.linalg.LinAlgError:
+            warnings.warn(
+                f"Inner solver of Newton solver {self.__class__.__name__} stumbbled "
+                "upon a singular matrix. Using SVD based least-squares solution "
+                "instead."
+            )
```

Comment:
I extended this code path quite a bit. On possibility where this can happen is if `X` is singular, like having collinear columns, and if the regularization is tiny or zero. I should write a test for it.